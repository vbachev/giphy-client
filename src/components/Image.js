import React from 'react'
import PropTypes from 'prop-types'

// @TODO:
// - lazy-load when in viewport
// - display title, user avatar and user name on hover

const Image = ({ url, width, height, title }) => {
  return (
    <div className='image-container'>
      <img className='image' src={url} width={width} height={height} alt={title} />
    </div>
  )
}

const { string, number } = PropTypes
Image.propTypes = {
  url: string.isRequired,
  width: string,
  heigt: string,
  title: string.isRequired
}

export default Image
