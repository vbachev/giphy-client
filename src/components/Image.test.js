import React from 'react'
import renderer from 'react-test-renderer'
import Image from './Image'

test('it renders an image', () => {
  const component = renderer.create(
    <Image url='https://google.com' title='Google Inc' width='100' height='60' />
  )
  expect(component.toJSON()).toMatchSnapshot()
})
