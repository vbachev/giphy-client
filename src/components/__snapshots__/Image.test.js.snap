// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders an image 1`] = `
<div
  className="image-container"
>
  <img
    alt="Google Inc"
    className="image"
    height="60"
    src="https://google.com"
    width="100"
  />
</div>
`;
